relay/ethToElrond/steps/stepsFactory.go

```diff
@@ -12,18 +13,44 @@ func CreateSteps(executor BridgeExecutor) (relay.MachineStates, error) {
 		return nil, ErrNilBridgeExecutor
 	}
 
-	return createMachineStates(executor), nil
+	return createMachineStates(executor)
 }
 
-func createMachineStates(executor BridgeExecutor) relay.MachineStates {
+func createMachineStates(executor BridgeExecutor) (relay.MachineStates, error) {
 	machineStates := make(relay.MachineStates)
 
-	machineStates[ethToElrond.GetPending] = &getPendingStep{
-		bridge: executor,
+	steps := []relay.Step{
```

Comment:
maybe declare this inside CreateSteps and pass it as an argument to createMachineStates